### PR TITLE
Nifi 3773

### PR DESCRIPTION
**Pull request by thesolson**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-docs/src/main/asciidoc/user-guide.adoc [hunk 1]
```diff
@@ -430,6 +430,41 @@ The Label has no text when initially created. The text of the Label can be added
 choosing `Configure`
 
 
+[[component-versioning]]
+=== Component Versions
+You have access to information about the version of your Processors, Controller Services, and Reporting Tasks.
+This is especially useful when you are working within a clustered environment with multiple NiFi instances running
+different versions of a component or if you have upgraded to a newer version of a processors. The Add Processor,
```
Comment (alopresto) [inline nifi-docs/src/main/asciidoc/user-guide.adoc]: ... newer version of a *processor* ...

Comment (andrewmlim) [inline nifi-docs/src/main/asciidoc/user-guide.adoc]: change "processors" to "processor"

File: nifi-docs/src/main/asciidoc/user-guide.adoc [hunk 2]
```diff
@@ -430,6 +430,41 @@ The Label has no text when initially created. The text of the Label can be added
...
+Add Controller Service, and Add Reporting Task dialogs include a column identifying the component version, as well
+as name of the component, the organization or group that created the component, and the NAR bundle that contains
```
Comment (alopresto) [inline nifi-docs/src/main/asciidoc/user-guide.adoc]: ... as well as *the* name of the component ...

File: nifi-docs/src/main/asciidoc/user-guide.adoc [hunk 3]
```diff
@@ -430,6 +430,41 @@ The Label has no text when initially created. The text of the Label can be added
...
+the processor. Each component displayed on the canvas also contains this information.
+
+==== Filtering Components
+When you are adding a component, you can filter based on originating source or version number.
+
+To filter based on version, click the version column to display in ascending or descending version order.
```
Comment (alopresto) [inline nifi-docs/src/main/asciidoc/user-guide.adoc]: This seems more like *sorting* than *filtering*.

File: nifi-docs/src/main/asciidoc/user-guide.adoc [hunk 4]
```diff
@@ -430,6 +430,41 @@ The Label has no text when initially created. The text of the Label can be added
...
+. Right-click the component on the canvas to display configuration options.
+. Select Change version.
+. In the Component Version dialog, select the version you want to run from the Version drop-down menu.
+
+image::component-version-dialog.png["Component Version"]
+
+==== Understanding Version Dependencies
+When you are configuring a component, you can also view information about version dependencies.
+
+. Right-click your component and select Configure to display the Configure dialog for your component
```
Comment (alopresto) [inline nifi-docs/src/main/asciidoc/user-guide.adoc]: Needs trailing '.'.

**Comment by alopresto** [comment]
Hi @thesolson . @andrewmlim and I made some suggestions. If you are ok with these or have other comments, please add them. I'll then merge this. Thanks for the great contribution.

File: nifi-docs/src/main/asciidoc/user-guide.adoc (hunk 3 above)
Comment (andrewmlim) [inline nifi-docs/src/main/asciidoc/user-guide.adoc]: The columns sort, the Source drop-down filters.  So maybe change the section to "Sorting and Filtering Components"

File: nifi-docs/src/main/asciidoc/user-guide.adoc (hunk 4 above)
Comment (andrewmlim) [inline nifi-docs/src/main/asciidoc/user-guide.adoc]: Add a period at the end of sentence:  "...for your component."

**Comment by alopresto** [comment]
With (out of band) confirmation from @thesolson , I've made the suggested changes and merged this.

**Comment by thesolson** [comment]
Thanks Andy, thanks Drew. Changes look good to me! Sarah Olson m: 415 298 5573 

    On Tuesday, May 2, 2017 2:38 PM, Andy LoPresto <notifications@github.com> wrote:
 

 With (out of band) confirmation from @thesolson , I've made the suggested changes and merged this.—

**Pull request closed without merging**